src/Swarm/Language/Syntax.hs

```diff
@@ -361,10 +375,67 @@ constInfo c = case c of
   commandLow = command (lowShow c)
   functionLow = function (lowShow c)
 
+mkOp' :: Const -> Term -> Term -> Term
+mkOp' c t1 t2 = TApp (TApp (TConst c) t1) t2
+
 -- | Make infix operation (e.g. @2 + 3@) a curried function
 --   application (@((+) 2) 3@).
-mkOp :: Const -> Term -> Term -> Term
-mkOp c = TApp . TApp (TConst c)
+mkOp :: Const -> Syntax -> Syntax -> Syntax
+mkOp c s1@(Syntax l1 _) s2@(Syntax l2 _) = Syntax newLoc newTerm
+ where
+  -- The new syntax span both terms
+  newLoc = Location (locStart l1) (locEnd l2)
```

Comment:
Use a `Semigroup` instance for `Location` here.  The code as-is assumes that `s1` came before `s2`, which is probably always true now but seems an unnecessary assumption.  If we make a generic, symmetric `Semigroup` instance then the order does not matter.

Reply:
Good point, thanks!